pkg/rules/kafka-go/kafka_otel_instrumenter.go

```diff
@@ -0,0 +1,322 @@
+	return "kafka"
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetDestination(request kafkaConsumerReq) string {
+	return request.msg.Topic
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetDestinationTemplate(request kafkaConsumerReq) string {
+	return ""
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) IsTemporaryDestination(request kafkaConsumerReq) bool {
+	return false
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) isAnonymousDestination(request kafkaConsumerReq) bool {
+	return false
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetConversationId(request kafkaConsumerReq) string {
+	return ""
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetMessageBodySize(request kafkaConsumerReq) int64 {
+	// Calculate message body size: 4(topic length) + 1(compression type) + 1(attributes) + 4(key length) + key length + 4(value length) + value length + 8(timestamp)
+	return int64(4 + 1 + 1 + 4 + len(request.msg.Key) + 4 + len(request.msg.Value) + 8)
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetMessageEnvelopSize(request kafkaConsumerReq) int64 {
+	return 0
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetMessageId(request kafkaConsumerReq, response any) string {
+	return ""
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetClientId(request kafkaConsumerReq) string {
+	return ""
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetBatchMessageCount(request kafkaConsumerReq, response any) int64 {
+	return 1
+}
+
+func (getter kafkaMessageConsumerAttrsGetter) GetMessageHeader(request kafkaConsumerReq, name string) []string {
+	if request.msg.Headers == nil {
+		return nil
+	}
+	var headerValues []string
+	for _, header := range request.msg.Headers {
+		if header.Key == name {
+			headerValues = append(headerValues, string(header.Value))
+		}
+	}
+	return headerValues
+}
+
+// KafkaProducerAttributesExtractor extracts producer attributes
+type kafkaProducerAttributesExtractor struct {
+}
+
+func (extractor *kafkaProducerAttributesExtractor) OnStart(attributes []attribute.KeyValue, parentContext context.Context, request kafkaProducerReq) ([]attribute.KeyValue, context.Context) {
+	kafkaAttributes := []attribute.KeyValue{
+		semconv.MessagingSystemKafka,
+		semconv.MessagingDestinationNameKey.String(request.topic),
+		semconv.MessagingOperationName("publish"),
+	}
+	return append(attributes, kafkaAttributes...), parentContext
+}
+
+func (extractor *kafkaProducerAttributesExtractor) OnEnd(attributes []attribute.KeyValue, ctx context.Context, request kafkaProducerReq, response any, err error) ([]attribute.KeyValue, context.Context) {
+	if err != nil {
+		attributes = append(attributes, attribute.String("error", err.Error()))
+	}
+	attributes = append(attributes, attribute.Bool("messaging.kafka.async", request.async))
+	attributes = append(attributes, attribute.String("messaging.kafka.broker_address", request.addr.String()))
```

Comment:
As far as I know, OpenTelemetry Kafka does not have those attributes in the latest spec(https://opentelemetry.io/docs/specs/semconv/messaging/kafka/). Do you follow the old semconv or just add those attributes by yourself?